WME-send-to-slack.user.js

```diff
@@ -530,67 +545,84 @@ ${closureTelegramDetails}${telegramDetails}`;
-                            N2: "②",
-                            N3: "③",
-                            N4: "④",
-                            N5: "⑤",
-                            N6: "⑥",
-                            L1: "①",
-                            L2: "②",
-                            L3: "③",
-                            L4: "④",
-                            L5: "⑤",
-                            L6: "⑥",
-                            ":": "*"
-                        };
-                        var discrordreplacefrom = new RegExp(Object.keys(discrordreplaceto).join("|"), "gi");
-                        $.ajax({
-                            data: 'payload=' + JSON.stringify({
-                                "attachments": [{
-                                    "text": TextToSend.replace(discrordreplacefrom, function(matched) {
-                                        return discrordreplaceto[matched]
-                                    })
-                                }],
-                                "username": ScriptName + " " + GM_info.script.version,
-                                "mrkdwn": true,
-                                "channel": serverDB[localStorage.getItem('WMESTSServer')][key]["chanel_" + chanel],
-                                "icon_emoji": actionicon
-                            }),
-                            processData: false,
-                            type: 'POST',
-                            url: serverDB[localStorage.getItem('WMESTSServer')][key][chanel],
-                            error: function(x, y, z)
-                            {
-                                log('Discord error : ' + x + ' ' + y + ' ' + z);
-                            }
-                        });
-                    }
+                    $.ajax({
+                        data: 'payload=' + JSON.stringify({
+                            "text": TextToSend,
+                            "username": ScriptName + " " + GM_info.script.version,
+                            "mrkdwn": true,
+                            "channel": serverDB[localStorage.getItem('WMESTSServer')][key]["chanel_" + chanel],
+                            "icon_emoji": actionicon
+                        }),
+                        processData: false,
+                        type: 'POST',
+                        url: serverDB[localStorage.getItem('WMESTSServer')][key][chanel],
+                        error: function (x, y, z) {
+                            log('Slack error : ' + x + ' ' + y + ' ' + z);
+                        }
+                    });
                     log(TextToSend);
                     sent=sent+1;
                     break;
-                case "gform":
+                    case "discord":
+                        log('Channel : ' + chanel);
+                        var actionicon = "";
+                        log(iconaction);
+                        switch(iconaction.toLowerCase()) {
```

Comment:
Not a fan that this whole switch is duplicated now, maybe extract it into a function

Reply:
the block can be moved in front of the outer switch block _switch (key.toLowerCase()) {_

Reply:
done